components/contribution-flow/index.js

```diff
@@ -446,8 +445,6 @@ class ContributionFlow extends React.Component {
       return false;
     } else if (this.props.tier?.type === TierTypes.TICKET) {
       return false;
-    } else if (this.state.stepProfile?.type === CollectiveType.COLLECTIVE) {
-      return this.state.stepProfile.host?.id && this.state.stepProfile.host.id === this.props.host?.id;
```

Comment:
I agree, I don't get what it is. Basically, it's too late to check canHaveFeesOnTop (step 1) if stepProfile is set (set at step2, available in step 3).

But there has to be a rationale.

Reply:
@Betree do you have more context on this? Is it safe to delete?

Reply:
Sorry for the late reply 😧 The rational seemed to be, when contributing from collective to collective, to only allow fees on top when contributing to another collective under the same host.

But it seems like a leftover copy paste that hasn't been adapted, because as @znarf said the steps order doesn't allow that check to work (it was different in the previous version of the flow), and cross-host collective to collective donations are not allowed.

So I think it's safe to delete!